Automattic/wp-calypso
# Hosting Signup: Add preference when trial plan selected

This PR fixes issue: https://github.com/Automattic/dotcom-forge/issues/6007

The hosting flow has several steps. The main issue here is that we don't know where to redirect the user to the hosting flow after the user has verified their email.

The patch D141645-code redirects the user after email verification.

This PR uses the calyspo preferences (`/me/preferences`) to save where in the hosting flow the user was last at when their email is not verified. 

### Test instructions
* Apply patch D141645-code to sandbox
* Sandbox the `public-api.wordpress.com` and `signup.wordpress.com` domains
* Apply this PR
* Go to http://calypso.localhost:3000/start/hosting
* Enter a test email account
<img width="553" alt="Create account with Email" src="https://github.com/Automattic/wp-calypso/assets/5560595/7ce153fc-bc99-4df9-b354-8e310760742e">

* There are 2 flows to choose from - Create a new site or Import site
<img width="628" alt="Hosting flow - 2 options" src="https://github.com/Automattic/wp-calypso/assets/5560595/69ca85b9-f43a-42aa-9526-422ba7b753de">

* To verify email address, find the email sent to the test email account. Copy the link into the browser you are using to verify the email address and it should redirect you to the last place you are on the flow. The redirect will be on `wordpress.com` rather than calypso.localhost:3000 but it should still work.
<img width="588" alt="Email verification" src="https://github.com/Automattic/wp-calypso/assets/5560595/640016e7-896d-4e22-b996-76073dce2300">

* On the create a new site path, confirm that if you go to the plans step or if you click the free trial plan, the trial acknowledge step - if you verify email address, it will redirect you to the correct step.
<img width="908" alt="Create new site - plans" src="https://github.com/Automattic/wp-calypso/assets/5560595/3d08fb23-d560-4fc5-9599-026d64a5b932">
<img width="860" alt="Create new site - verify email to continue" src="https://github.com/Automattic/wp-calypso/assets/5560595/314fe0df-a2af-493f-8832-dceb2ae4218e">

* On the import site path, confirm that if you go to the plans step or if you click the free trial plan, the verify email step
<img width="734" alt="Import site - plans" src="https://github.com/Automattic/wp-calypso/assets/5560595/249031da-8993-40d5-828d-950606808810">
<img width="625" alt="Import site - verify email to continue" src="https://github.com/Automattic/wp-calypso/assets/5560595/0daa8361-b185-4a69-b274-06b586c31534">
